device-server/src/main/kotlin/com/badoo/automation/deviceserver/ios/fbsimctl/FBSimctlResponseParser.kt

```diff
@@ -71,6 +71,35 @@ class FBSimctlResponseParser : IFBSimctlResponseParser {
         return parsedResponse.subject
     }
 
+    override fun parseInstallApp(response: String): FBSimctlInstallResult {
+        val mapper = JsonMapper()
+        val events = response.lines()
+            .asSequence()
+            .filter { !isLogEvent(it) }
+            .map { mapper.fromJson<Map<String, Any>>(it) }
+            .toList()
+
+        var errorMessage = ""
+
+        val isInstalled = events.any {
+            it["event_type"] == "ended" && it["event_name"] == "install"
+        }
+
+        if (!isInstalled) {
+            val failureEvent = events.find {
```

Comment:
Are you sure that `failure` event cannot be present if `install` event has `ended`?

Reply:
Good point. fixed a bit. now if there is one, we'll get it. Thank you!